### ISSUE TITLE
Add Login-Signup Feature with Authentication

### ISSUE THREAD

**Issue by saras-69** [GarimaSingh0109/WasteManagment#308]
### Description

This feature aims to implement a secure login and signup system with user authentication. Users will be able to create accounts, log in, and access restricted areas of the application based on their authentication status. Alternative solutions, such as third-party authentication services , have been considered but are not part of this initial implementation.

### Motivation

Implementing a login-signup feature enhances user experience by allowing personalized access to the application. It improves security by ensuring that only authorized users can access specific functionalities and keeps user data secure.

### Additional Context

This feature will be built using standard authentication practices, including input validation, password hashing, and secure session management. It may also include options for password recovery and user role management in future iterations.

### Checklist

- [X] I have checked the existing [issues](https://github.com/GarimaSingh0109/WasteManagment/issues).
- [X] I have read the [Code Of Conduct](https://github.com/GarimaSingh0109/WasteManagment?tab=coc-ov-file).
- [X] The changes don't break the code.
- [X] I will attach screenshots and videos of the development work once it is done.

### Priority

High
